### PR TITLE
[kf5sonnet] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**

- #### What does your PR fix?  
  Adds new port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official triplets.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

### CODE REVIEW

**Comment by wrobelda** [comment]
@hellozee much of it was your work, so please test again on your end if possible

**Comment by hellozee** [comment]
> @hellozee much of it was your work, so please test again on your end if possible

Ahh sure, 😺

File: ports/kf5sonnet/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO KDE/sonnet
+    REF v5.84.0
+    SHA512 9e7d121f447e3320c27c3708f5d1d4cc735e775749cded268502b593a0b1f6ea703e68ce1d2d4f1806e0adb73aafaedf660586f8ee740f4a9a834e23cb9880e4
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (PhoebeHui) [inline ports/kf5sonnet/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/kf5sonnet/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    MAYBE_UNUSED_VARIABLES
+        -DBUILD_HTML_DOCS=OFF
+        -DBUILD_MAN_DOCS=OFF
+        -DBUILD_QTHELP_DOCS=OFF
+)
+
+vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}/bin")
+vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}/debug/bin")
+
+vcpkg_install_cmake()
```
Comment (PhoebeHui) [inline ports/kf5sonnet/portfile.cmake:26]:
```suggestion
vcpkg_cmake_install()
```

File: ports/kf5sonnet/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/gentrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/gentrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/parsetrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/parsetrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/KF5Sonnet)
```
Comment (PhoebeHui) [inline ports/kf5sonnet/portfile.cmake:32]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/KF5Sonnet)
```

File: ports/kf5sonnet/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/etc")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/bin/gentrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/bin/parsetrigrams${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+
+file(INSTALL ${SOURCE_PATH}/LICENSES/ DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
```
Comment (PhoebeHui) [inline ports/kf5sonnet/portfile.cmake:49]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSES" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/kf5sonnet/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "kf5sonnet",
+  "version-string": "5.84.0",
+  "description": "Multi-language spell checker",
+  "homepage": "https://api.kde.org/frameworks/sonnet/html/index.html",
+  "dependencies": [
+    "ecm",
+    "qt5-base",
+    "qt5-tools"
```
Comment (PhoebeHui) [inline ports/kf5sonnet/vcpkg.json:9]:
```suggestion
    "qt5-tools",
     {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@wrobelda, thanks for your PR!

These functions has been deprecated:  [vcpkg_configure_cmake](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md), [vcpkg_install_cmake](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_cmake.md) and [vcpkg_fixup_cmake_targets](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_fixup_cmake_targets.md)

File: ports/kf5sonnet/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "kf5sonnet",
+  "version-string": "5.84.0",
```
Comment (PhoebeHui) [inline ports/kf5sonnet/vcpkg.json:3]:
```suggestion
  "version-semver": "5.84.0",
```

File: ports/kf5sonnet/portfile.cmake (hunk 4 above)
Comment (wrobelda) [reply ports/kf5sonnet/portfile.cmake]: `${SOURCE_PATH}/LICENSES/` is a folder, not a file, and it contains several SPDX license files which all need to be copied to "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" folder.

Comment (PhoebeHui) [reply ports/kf5sonnet/portfile.cmake]: I see, thanks for clarifying this!

**Comment by wrobelda** [comment]
> @wrobelda, thanks for your PR!
> 
> These functions has been deprecated: [vcpkg_configure_cmake](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md), [vcpkg_install_cmake](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_cmake.md) and [vcpkg_fixup_cmake_targets](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_fixup_cmake_targets.md)

Thanks! Totally forgot about it. All fixed now.

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Pull request merged**